core/src/indexer/native_transfer.rs

```diff
@@ -0,0 +1,130 @@
+use std::{str::FromStr, sync::Arc};
+
+use colored::Colorize;
+use ethers::{
+    abi::Hash,
+    types::{Action, Address, Bytes, U256, U64},
+};
+use futures::future::try_join_all;
+use serde::Serialize;
+use serde_json::json;
+use tracing::{error, info};
+
+use super::start::StartIndexingError;
+use crate::{
+    event::{callback_registry::TxInformation, EventMessage},
+    provider::JsonRpcCachedProvider,
+    streams::StreamsClients,
+};
+
+#[derive(Serialize)]
+pub struct NativeTransfer {
+    pub from: Address,
+    pub to: Address,
+    pub value: U256,
+    pub transaction_information: TxInformation,
+}
+
+pub async fn native_transfer_block_consumer(
+    provider: Arc<JsonRpcCachedProvider>,
+    block_numbers: &[U64],
+    network_name: &str,
+    streams_client: Option<&StreamsClients>,
+) -> Result<(), StartIndexingError> {
+    let trace_futures: Vec<_> = block_numbers.iter().map(|n| provider.trace_block(*n)).collect();
+
+    let trace_calls = try_join_all(trace_futures).await?;
+
+    let native_transfers = trace_calls
+        .into_iter()
+        .flatten()
+        .filter_map(|trace| {
+            let action = match trace.action {
+                Action::Call(call) => Some(call),
+                _ => None,
+            }?;
+
+            let has_value = !action.value.is_zero();
+            let no_input = action.input == Bytes::from_str("0x").unwrap();
```

Comment:
cant you do 

```
let no_input = action.input == Bytes::new(); 
```

Reply:
not sure what input is but probs nicer way to do it then bringing in an unwrap

Reply:
Hmm, probably can yeah. Let me try